src/dualscreen.tsx

```diff
@@ -251,8 +251,10 @@ export class DualScreen extends BasePlugin<DualScreenConfig> implements IEngineD
         const videoTrack = dualScreenPlayer.player.getTracks('video');
         const selectedVideoTrack = videoTrack.find((track: any) => track.label === label);
         if (selectedVideoTrack) {
-          //@ts-expect-error
-          dualScreenPlayer.player.changeQuality(selectedVideoTrack);
+          setTimeout(()=> {
+            //@ts-expect-error
+            dualScreenPlayer.player.changeQuality(selectedVideoTrack);
+          }, 1000);
```

Comment:
setTimeout might cause inconsistent behaviour despite of network, cpu, memory, etc...
I would recommend find and use player event then trigger `dualScreenPlayer.player.changeQuality` once the event dispatched.